### ISSUE TITLE
Not handling pytorch memory allocation right on an Mac

### ISSUE THREAD

**Issue by To3No7** [mittagessen/kraken#801]
I installed Kraken on a MacBook Pro (M4) with
`pipx install --python python3.13 kraken`
... installed package kraken 7.1, installed using Python 3.13.1

And get the following error when running
`kraken -i IMG20260813142537.jpg output.txt segment -bl ocr -m $MODELPATH/medium.safetensors `

Segmenting IMG20260813142537.jpg	✓
Processing ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% 0/13 -:--:-- 0:00:01
[09/04/26 16:22:43] ERROR    Failed processing IMG20260813142537.jpg: Expected all tensors to be on the same device, but found at least two devices, mps:0 and cpu!  kraken.py:424

**Comment by mittagessen** [comment]
This bug was fixed a couple of weeks ago but I haven't tagged a new
release yet. Installing main should run everything correctly on MPS (or
you can run inference on cpu with the installed version by specifying
`kraken -d cpu ...`

**Comment by To3No7** [comment]
Some Google Gemini analysis:
### Cause of the Error

  When running Kraken on Apple Silicon (M4), PyTorch Lightning's Fabric automatically detects
  Apple Silicon GPU acceleration (mps) when the -d / --device flag is omitted or set to auto.
  Fabric moves the neural network model parameters (self.nn) onto mps:0.

  However, during recognition inference in ctc.py:231, the input line tensors (line and lens)
  extracted from the image remain on the cpu. When PyTorch performs the forward pass
  self.nn(line, lens), it raises:
  Expected all tensors to be on the same device, but found at least two devices, mps:0 and cpu!

### Solution: Codebase Fix (Enable MPS GPU acceleration)

  To support mps GPU execution without device mismatch errors, input tensors must be explicitly
  moved to self._fabric.device before the model forward pass.

  #### Relevant Locations in Codebase:

  1. **ctc.py:216-235** in _rec_predict:

    line = line.to(self._fabric.device)
    if lens is not None:
        lens = lens.to(self._fabric.device)
    logits, olens = self.nn(line, lens)

  2. **spred.py:268** in _compute_segmentation_map:

    o, _ = self.nn(tensor_im.unsqueeze(0).to(self._fabric.device))

**Comment by mittagessen** [comment]
As I said, it's already been fixed on August 15.
